src/OrchardCore.Modules/OrchardCore.Seo/Migrations.cs

```diff
@@ -37,7 +39,14 @@ public int Create()
                     .WithSettings(new MediaFieldSettings { Multiple = false }))
             );
 
-            return 1;
+            SchemaBuilder.CreateMapIndexTable<SeoMetaPartIndex>(table => table
+                .Column<string>("PageTitle", c => c.WithLength(128)));
+
+            SchemaBuilder.AlterIndexTable<SeoMetaPartIndex>(table => table
+                .CreateIndex("IDX_SeoMetaPartIndex_DocumentId", "DocumentId", "PageTitle")
+            );
+
+            return 3;
```

Comment:
You need to keep return 1 here and add a UpdateFrom2Async() method which will do the same as the lines that you added here.

Reply:
Just for info, yes we need UpdateFromX migration steps, but normally the rule is to also update the Create() to do everything and then skip these steps, so that on a fresh install all things can be done in one pass without having e.g. to add/remove/re-add/re-defining things.

Here looks like this pattern was not used, so would need to also add in Create() what UpdateFrom1 does, then add what a new UpdateFrom2 (needs to be defined) will do, and return 3 to skip these steps.

Reply:
I have a question here as to when the Migration is actually triggered? I setup everything but the UpdateFromX aren't triggered. I tried disabling the module and reenabling it but still no call, what's the rule behind this one? Is there some kind of versioning somewhere?

Reply:
When a migration method is executed the return value of the method is stored in the database in the document table in a single row containing all the migration versions that have been executed for each feature. We can't reset the versioning.

Reply:
I see, thanks, I added the suggested parts and left the return 3 since it does everything it needs to do now